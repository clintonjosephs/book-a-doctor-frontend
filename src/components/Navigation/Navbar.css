.navbar {
  position: fixed;
  overflow: auto;
}

.menu-bars {
  margin-left: 2rem;
  font-size: 2rem;
  background: none;
}

.nav-menu {
  width: 280px;
  height: 100vh;
  display: flex;
  justify-content: center;
  position: fixed;
  top: 0;
}

.nav-menu.active {
  left: 0;
  transition: 350ms;
}

.nav-text {
  display: flex;
  justify-content: start;
  align-items: center;
  padding: 8px 0 8px 16px;
  list-style: none;
  height: 60px;
}

.nav-text a {
  text-decoration: none;
  color: #000;
  font-size: 18px;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  padding: 0 16px;
  border-radius: 4px;
  cursor: pointer;
}

.nav-text a:hover {
  background-color: #97bf0f;
}

.nav-menu-items {
  width: 100%;
}

.user-info {
  width: 100%;
  height: 30vh;
  display: flex;
  flex-direction: column;
  gap: 10px;
  justify-content: center;
  align-items: center;
}

.show-menu {
  display: none;
}

.close-menu {
  display: none;
}

.user-name {
  font-size: 1.5rem;
}

.image {
  width: 100px;
  height: 100px;
  border-radius: 50%;
}

span {
  margin-left: 16px;
}

.logout {
  border: 2px solid red;
  width: 70%;
  height: 40px;
  margin: 20px;
  font-size: 20px;
  color: red;
  cursor: pointer;
  bottom: 0;
  position: absolute;
}

.social-menu {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 90%;
  gap: 20px;
  margin-top: 30px;
}

.social-icons {
  width: 25px;
  height: 25px;
  cursor: pointer;
}

.logout:hover {
  background-color: red;
  transform: scale(1.1);
  color: #fff;
}

a.active {
  background-color: #ffb400;
}

.navbar-toggle {
  width: 100%;
  display: none;
}

@media (max-width: 992px) {
  .show-menu {
    display: flex;
  }

  .nav-menu-items {
    width: 100%;
    background-color: #ffb400;
  }

  .user-info {
    height: 22vh;
  }

  .navbar-toggle {
    display: flex;
    justify-content: end;
    align-items: center;
  }

  .nav-menu {
    left: -100%;
    width: 100%;
  }

  .social-menu {
    display: flex;
    justify-content: start;
    margin-left: 30px;
    width: 100%;
    gap: 40px;
  }

  .logout {
    margin: 20px 40px;
  }

  .nav-menu ~ div section {
    z-index: 100;
  }
  
  .nav-menu.active ~ div section {
    z-index: -100;
  }
  
}
