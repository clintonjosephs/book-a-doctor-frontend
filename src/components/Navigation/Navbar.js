import React, { useState } from 'react';
import * as FaIcons from 'react-icons/fa';
import { NavLink, useNavigate } from 'react-router-dom';
import * as AiIcons from 'react-icons/ai';
import { IconContext } from 'react-icons';
import { useSelector } from 'react-redux';
import SidebarData from './SidebarData';
import './Navbar.css';
import StorageManager from '../../helpers/format/StorageManager';

function Navbar() {
  const [sidebar, setSidebar] = useState(false);

  const user = useSelector((state) => state.userReducer.userDetails);

  const showSidebar = () => setSidebar(!sidebar);
  const getClass = ({ isActive }) => (isActive ? 'active' : '');
  const navigate = useNavigate();
  const logout = () => {
    StorageManager.removeToken();
    navigate('/login');
  };

  return (
    <>
      {user
      && (
      <IconContext.Provider value={{ color: '#000' }}>
        <div className="navbar">
          {!sidebar && (
          <div className="menu-bars">
            <FaIcons.FaBars onClick={showSidebar} className="show-menu" />
          </div>
          ) }
        </div>
        <nav className={sidebar ? 'nav-menu active' : 'nav-menu'}>
          <ul className="nav-menu-items" onClick={showSidebar} onKeyDown={showSidebar} role="presentation">
            <li className="navbar-toggle">
              <div className="menu-bars">
                <AiIcons.AiOutlineClose className="close=menu" />
              </div>
            </li>
            <li className="user-info">
              <img src={user.image_url} alt="user" className="image"/>
              <p className="user-name">{user.name}</p>
            </li>
            {SidebarData(user.role).map((data) => (
              <li key={data.id} className={data.className}>
                <NavLink to={data.path} className={getClass}>
                  {data.icon}
                  <span>{data.title}</span>
                </NavLink>
              </li>
            ))}
            <div className="social-menu">
              <FaIcons.FaFacebook className="social-icons" />
              <FaIcons.FaTwitter className="social-icons" />
              <FaIcons.FaInstagram className="social-icons" />
              <FaIcons.FaGithubAlt className="social-icons" />
            </div>
            <button type="button" className="logout" onClick={logout}>Logout</button>
          </ul>
        </nav>
      </IconContext.Provider>
      )}
    </>
  );
}

export default Navbar;
