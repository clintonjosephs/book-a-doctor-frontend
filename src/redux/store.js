import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import bookDoctorReducer from './bookDoctor/doctorActions';
import userReducer from './user/userActions';
import reservationReducers from './reservation/reservationActions';

const reducer = combineReducers({
  bookDoctorReducer,
  userReducer,
  reservationReducers,
});

// creating a store
const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
