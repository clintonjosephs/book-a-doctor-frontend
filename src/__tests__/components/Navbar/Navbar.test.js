import {
  render, screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../../../components/Navigation/Navbar';
import store from '../../../redux/store';

const Wrapper = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  </Provider>
);

describe('Navbar Component', () => {
  it('should render link if user is  logged in', () => {
    render(<Wrapper />);

    const Doctors = screen.getByText('Doctors');
    const Reservations = screen.getByText('Reservations');
    const BookAppointments = screen.getByText('Book Appointments');
    const userImage = screen.getByAltText('user');
    expect(Doctors).toMatchSnapshot();
    expect(Reservations).toMatchSnapshot();
    expect(BookAppointments).toMatchSnapshot();
    expect(userImage).toBeInTheDocument();
  });

  it('it should render logout', () => {
    render(<Wrapper />);
    const Logout = screen.getByText('Logout');

    expect(Logout).toBeInTheDocument();
  });
});
