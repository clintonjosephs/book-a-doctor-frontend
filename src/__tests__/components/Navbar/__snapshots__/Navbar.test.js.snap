// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render link if user is  logged in 1`] = `
<span>
  Doctors
</span>
`;

exports[`Navbar Component should render link if user is  logged in 2`] = `
<span>
  Reservations
</span>
`;

exports[`Navbar Component should render link if user is  logged in 3`] = `
<span>
  Book Appointments
</span>
`;
